fuel-core/src/cli/run/p2p.rs

```diff
@@ -131,75 +132,80 @@ pub struct P2pArgs {
     pub connection_keep_alive: u64,
 }
 
-impl From<P2pArgs> for anyhow::Result<P2PConfig> {
-    fn from(args: P2pArgs) -> Self {
+impl P2pArgs {
+    pub fn into_config(self, chain_config: &ChainConfig) -> anyhow::Result<P2PConfig> {
+        let checksum = *fuel_crypto::Hasher::default()
```

Comment:
Hmm this fn you linked to doesn't include the initial state root or chain name though

Reply:
Then, maybe better to use the root of the genesis block=)

Regarding the name of the network, we can include it, if you think that we need this kind of information in the `ChainConfig`

Reply:
Yeah it should be a checksum of the full chainspec / entire genesis block.

Also, these root functions should exist in the chain-config module so that they can be used without fuel-core. As fuel-p2p is in it's own crate and won't be able to use the genesis module directly.